topic/chatbot/dis2025/workshop/Timeseries_QA_Workshop.ipynb

```diff
@@ -0,0 +1,1837 @@
+      "Requirement already satisfied: parso<0.9.0,>=0.8.3 in /Users/wierd/Library/Python/3.9/lib/python/site-packages (from jedi>=0.16->ipython->ipython-sql) (0.8.3)\n",
+      "Requirement already satisfied: ptyprocess>=0.5 in /Users/wierd/Library/Python/3.9/lib/python/site-packages (from pexpect>4.3->ipython->ipython-sql) (0.7.0)\n",
+      "Requirement already satisfied: executing>=1.2.0 in /Users/wierd/Library/Python/3.9/lib/python/site-packages (from stack-data->ipython->ipython-sql) (2.0.1)\n",
+      "Requirement already satisfied: asttokens>=2.1.0 in /Users/wierd/Library/Python/3.9/lib/python/site-packages (from stack-data->ipython->ipython-sql) (2.4.1)\n",
+      "Requirement already satisfied: pure-eval in /Users/wierd/Library/Python/3.9/lib/python/site-packages (from stack-data->ipython->ipython-sql) (0.2.2)\n",
+      "\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m23.3.1\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m25.0.1\u001b[0m\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49m/Library/Developer/CommandLineTools/usr/bin/python3 -m pip install --upgrade pip\u001b[0m\n",
+      "Defaulting to user installation because normal site-packages is not writeable\n",
+      "Requirement already satisfied: crate in /Users/wierd/Library/Python/3.9/lib/python/site-packages (0.34.0)\n",
+      "Requirement already satisfied: urllib3<2.1 in /Users/wierd/Library/Python/3.9/lib/python/site-packages (from crate) (1.26.18)\n",
+      "\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m23.3.1\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m25.0.1\u001b[0m\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49m/Library/Developer/CommandLineTools/usr/bin/python3 -m pip install --upgrade pip\u001b[0m\n",
+      "Defaulting to user installation because normal site-packages is not writeable\n",
+      "Requirement already satisfied: tqdm in /Users/wierd/Library/Python/3.9/lib/python/site-packages (4.66.5)\n",
+      "\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m23.3.1\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m25.0.1\u001b[0m\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49m/Library/Developer/CommandLineTools/usr/bin/python3 -m pip install --upgrade pip\u001b[0m\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Install dependencies\n",
+    "!pip install pandas matplotlib openai sqlalchemy python-dotenv ipython-sql\n",
```

Comment:
Excellent choice selecting `ipython-sql`. However, I can't spot any `%%sql` blocks, are you actually using it? On another occasion, you may want to try [JupySQL](https://jupysql.ploomber.io/)?

-- See examples with CrateDB here: https://github.com/crate/cratedb-examples/tree/main/notebook/jupyter